operator_tracker/src/lib.rs

```diff
@@ -0,0 +1,160 @@
+use axum::{body::Body, response::IntoResponse};
+use base64::prelude::BASE64_STANDARD;
+use base64::Engine;
+use ethers::providers::{Http, Provider};
+use ethers::utils::keccak256;
+use ethers::{contract::abigen, core::types::Signature};
+use log::{debug, info};
+use serde::Serialize;
+use sqlx::{Executor, PgPool};
+
+abigen!(
+    RegistryCoordinator,
+    r#"[function getOperatorStatus(address) external view returns (uint8)]"#,
+);
+
+pub type RegistryCoordinatorContract = RegistryCoordinator<Provider<Http>>;
+
+#[derive(serde::Deserialize)]
+pub struct OperatorVersionPayload {
+    pub version: String,
+    pub signature: String,
+}
+
+#[derive(serde::Serialize, Debug)]
+pub enum OperatorVersionError {
+    InvalidSignature,
+    OperatorNotRegistered,
+}
+
+impl IntoResponse for OperatorVersionError {
+    fn into_response(self) -> axum::http::Response<Body> {
+        axum::http::Response::builder()
+            .status(axum::http::StatusCode::BAD_REQUEST)
+            .body(Body::from(serde_json::to_string(&self).unwrap()))
+            .unwrap()
```

Comment:
let's not use unwraps, if there is any error while building the response body let's respond something